orangecontrib/spectroscopy/io/ptir.py

```diff
@@ -65,6 +67,42 @@ def read_spectra(self):
         wavenumbers = []
         x_locs = []
         y_locs = []
+        # for including focus information
+        z_locs = []
+
+        # map checked images in file
+        image_channels = []
+        if 'Heightmaps' in keys:
+            for img_name in [i for i in hdf5_file['Heightmaps'].keys()]:
+                if hdf5_file['Heightmaps'][img_name].attrs['Checked'][0] == 1:
+                    image_channels.append(img_name)
+        if 'Images' in keys:
+            for img_name in [i for i in hdf5_file['Images'].keys()]:
+                if hdf5_file['Images'][img_name].attrs['Checked'][0] == 1:
+                    image_channels.append(img_name)
```

Comment:
Ran into a case, where image in PTIR files does not have `'Checked'` attribute. To account for this issue:
```suggestion
                img_attrs = hdf5_file['Images'][img_name].attrs
                if img_attrs.keys().__contains__('Checked') and not img_attrs['Checked'][0]:
                    image_channels.append(img_name)
```
Or better to put into `try` to catch future cases?
```suggestion             
                try:
                    if hdf5_file['Images'][img_name].attrs['Checked'][0] == 1:
                        image_channels.append(img_name)
                except AttributeError:
                    warnings.warn(f"Unsupported visible image entry {img_name}")
```
Same would be needed earlier for `'Heightmaps'`

Reply:
The first is fine, but it should be `if "checked" in img_attrs.keys() and img_attrs['Checked'][0] == 1:`

(You changed the boolean logic from the original, was that intentional?)

Or maybe just `if hdf5_file['Images'][img_name].attrs.get('Checked', [None])[0] == 1:`

What is the "checked" parameter anyway?

Reply:
Ok thanks, will change accordingly.

Checked is parameter in the PTIR studio software that allows you to deselect spectra/images that should not be displayed. It can be useful to cut down on spectra/images you have used for finding the area of interest.

Reply:
Hmm, it seems strange to me in a file loader to respect gui selections from another software? Perhaps it would be more appropriate to load them all and then put the unchecked ones at the bottom of the list or something. But I'll defer to your experience on this one :)

Reply:
We discussed this offline and @kaiyang-tu preferred to leave it as is. Can be changed in future PR if needed.